crates/autopilot/src/on_settlement_event.rs

```diff
@@ -306,3 +216,224 @@ impl Inner {
         }
     }
 }
+
+#[async_trait]
+impl OnSettlementEventAction for Updater {
+    async fn apply(&self) -> Result<bool> {
+        let mut ex = self
+            .db
+            .pool
+            .begin()
+            .await
+            .context("acquire DB connection")?;
+        let event = match database::settlements::get_settlement_without_auction(&mut ex)
+            .await
+            .context("get_settlement_without_auction")?
+        {
+            Some(event) => event,
+            None => return Ok(false),
+        };
+
+        let hash = H256(event.tx_hash.0);
+        tracing::debug!("updating settlement details for tx {hash:?}");
+
+        let Ok(transaction) = self.eth.transaction(hash.into()).await else {
+            tracing::warn!(?hash, "no tx found");
+            return Ok(false);
+        };
+
+        let (auction_id, auction_data) =
+            match Self::recover_auction_id_from_calldata(&mut ex, &transaction).await? {
+                AuctionIdRecoveryStatus::InvalidCalldata => {
+                    // To not get stuck on indexing the same transaction over and over again, we
+                    // insert the default auction ID (0)
+                    (Default::default(), None)
+                }
+                AuctionIdRecoveryStatus::DoNotAddAuctionData(auction_id) => (auction_id, None),
+                AuctionIdRecoveryStatus::AddAuctionData(auction_id, settlement) => (
+                    auction_id,
+                    Some(
+                        self.fetch_auction_data(hash, settlement, auction_id)
+                            .await?,
+                    ),
+                ),
+            };
+
+        let update = SettlementUpdate {
+            block_number: event.block_number,
+            log_index: event.log_index,
+            auction_id,
+            auction_data,
+        };
+
+        tracing::debug!(?hash, ?update, "updating settlement details for tx");
+
+        Postgres::update_settlement_details(&mut ex, update.clone())
+            .await
+            .with_context(|| format!("insert_settlement_details: {update:?}"))?;
+        ex.commit().await?;
+        Ok(true)
+    }
+}
+
+#[allow(dead_code)]
+pub struct CircuitBreaker {
+    eth: infra::blockchain::circuit_breaker::CircuitBreaker,
+    persistence: infra::Persistence,
+    db: Postgres,
+    solvers: Vec<eth::Address>,
+    // Registry to keep track of the already processed settlements, format (block, hash)
+    registry: Arc<Mutex<BTreeMap<i64, H256>>>,
+}
+
+impl CircuitBreaker {
```

Comment:
CircuitBreaker as a component probably deserves to have it's own file somewhere under the`autopilot::domain`. Meaning, going forward, we should try to fit new functionalities under the domain/infra umbrella.

Reply:
@sunce86 Yes, my idea was to move this to the place it belongs once we have the implementation needed in `domain`. This is an incomplete PR since it needs follow up works, but that was my idea all along.